src/main/java/network/brightspots/rcv/CommonDataFormatReader.java

```diff
@@ -22,28 +22,91 @@
 
 package network.brightspots.rcv;
 
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlElementWrapper;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
 import java.io.File;
+import java.io.FileInputStream;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.logging.Level;
 import javafx.util.Pair;
+import network.brightspots.rcv.CastVoteRecord.CvrParseException;
 import network.brightspots.rcv.TabulatorSession.UnrecognizedCandidatesException;
 
 class CommonDataFormatReader {
 
+  private final String STATUS_NEEDS_ADJUDICATION = "needs-adjudication";
+  private final String STATUS_NO = "no";
+  private final String BOOLEAN_TRUE = "true";
+
   private final String filePath;
   private final ContestConfig config;
-  private Map<String, Integer> unrecognizedCandidateCounts = new HashMap<>();
+  private final String contestId;
+  private final Map<String, Integer> unrecognizedCandidateCounts = new HashMap<>();
 
-  CommonDataFormatReader(String filePath, ContestConfig config) {
+  CommonDataFormatReader(String filePath, ContestConfig config, String contestId) {
     this.filePath = filePath;
     this.config = config;
+    this.contestId = contestId;
+  }
+
+  // This method will extract candidate data from a CDF file for the contestID specified in
+  // our CvrSource.
+  Map<String, String> getCandidates() throws CvrParseException {
```

Comment:
I'm confused. The only place you call this is in `parseJson`, so when would it ever call `getCandidatesXml`? And that method doesn't have any other callsites, so I don't understand how it gets used.